sw/device/lib/base/hardened_status.h

```diff
@@ -28,17 +28,23 @@ extern "C" {
 /**
  * Hardened version of the `TRY` macro from `status.h`.
  *
+ * Returns the error unmodified if `status_ok` fails, and throws an
+ * `INTERNAL()` errror if the OK code does not match the hardened value.
+ *
  * @param expr_ An expression that evaluates to a `status_t`.
  * @return The enclosed OK value.
  */
-#define HARDENED_TRY(expr_)                                 \
-  ({                                                        \
-    status_t status_ = expr_;                               \
-    if (hardened_status_ok(status_) != kHardenedBoolTrue) { \
-      return status_;                                       \
-    }                                                       \
-    HARDENED_CHECK_EQ(status_.value, kHardenedBoolTrue);    \
-    status_.value;                                          \
+#define HARDENED_TRY(expr_)                              \
+  ({                                                     \
+    status_t status_ = expr_;                            \
+    if (!status_ok(status_)) {                           \
+      return status_;                                    \
+    }                                                    \
+    if (launder32(status_.value) != kHardenedBoolTrue) { \
+      return INTERNAL();                                 \
```

Comment:
Where does this definition of `INTERNAL` come from?  If it comes from `base/status.h`, it probably constructs the error code via `status_create()`.  You may not want that function in cryptolib.

You can construct this manually with the right set of shifts:
```
0x80000000 | MODULE_ID | ((__LINE__ & 0x7FF) << 5) | kInternal)
```
(although, you'll probably run afoul of the `bitfield.h` police</snark>).

Reply:
Yep, I agree -- I put some thoughts on that at the very end of the PR description, although now I see it was probably user-unfriendly to put it after a huge chunk of disassembly. Essentially, I'm wondering if maybe it makes sense to move the whole `hardened_status.h` into the cryptolib, since it's the only user anyway. And then we could return `OTCRYPTO_FATAL_ERR` or something here, which would have the nice property of giving us a module/line number when debugging is on and the hardened cryptolib error code when debugging is off.